relay/kusama/src/lib.rs

```diff
@@ -1839,7 +1839,7 @@ construct_runtime! {
 		ParaSessionInfo: parachains_session_info = 61,
 		ParasDisputes: parachains_disputes = 62,
 		ParasSlashing: parachains_slashing = 63,
-		OnDemandAssignmentProvider: parachains_on_demand = 64,
+		OnDemand: parachains_on_demand = 64,
```

Comment:
This rename will require a migration of the storage too. 

I was working on this on the past but at that point there were conflicting migrations. I don't remember the details but I can dig up my old PR.